service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -585,6 +585,28 @@ public void setPrivateResourcesStateForWorkspaceUser(
     jdbcTemplate.update(sql, params);
   }
 
+  /**
+   * Wait for a resource row to appear in the database. This is used so async resource creation
+   * calls return after the database row has been inserted. That allows clients to see the resource
+   * with a resource enumeration.
+   *
+   * @param workspaceId workspace where the resource is being created
+   * @param resourceId resource being created
+   */
+  @ReadTransaction
+  public boolean checkForResourceRow(UUID workspaceId, UUID resourceId) {
```

Comment:
Since it's a boolean function, I'd name it like `resourceExists()`. It's implied that we're talking about a DB row since we're in the Dao.